File: Easy/TwoSum/twoSum.py
from collections import defaultdict

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # for idx, num in enumerate(nums):
        #     next_idx = idx + 1

        #     if next_idx < len(nums):
        #         for next_num in nums[next_idx:]:
        #             if num + next_num == target:
        #                 return [idx, next_idx]
        #             next_idx += 1
                    
        # return False
        hashMap = defaultdict()
        for idx, num in enumerate(nums):
            try:
                value = hashMap[target-num]
                return [idx, value]
            except:
                hashMap[num] = idx

File: Easy/TwoSum/twoSum.go
package main

func twoSum(nums []int, target int) []int {
    hashMap := make(map[int]int)

    for i, num := range nums {
        if j, ok := hashMap[target-num]; ok {
            return []int{i,j}
        }
        hashMap[num] = i
    }
    return []int{-1,-1}
}

